src/holoscan_cli/utils/host_setup.py

```
@@ -210,21 +210,32 @@ def setup_cmake(min_version: str = "3.26.4", dry_run: bool = False) -> None:
+    keyring_path = "/usr/share/keyrings/kitware-archive-keyring.gpg"
+    source_line = (
+        "deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] "
+        f"https://apt.kitware.com/ubuntu/ {ubuntu_codename} main\n"
     )
+    if dry_run:
+        info(f"[dryrun] Would fetch the Kitware archive key -> {keyring_path}")
+        write_system_file("/etc/apt/sources.list.d/kitware.list", source_line, dry_run=True)
+    else:
+        # Fetch + dearmor the key as the invoking user; install the keyring as root.
+        try:
+            dearmored = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call with `shell=True` seems safe, but may be changed in the future; consider rewriting without `shell`

(S602)